better SDK documentation for DCL protocol (#129)
decentraland/adr: content/ADR-102-decentraland-protocol-for-explorers.md

> TODO: Write about the lifecycle of a scene and how it is loaded

### Loading scenes

> As explained avobe, scenes are programs written in JavaScript. The deployed scenes must comply with the Scene schema defined in [ADR-51](/adr/ADR-51). Entities can be loaded as scenes if their metadata matches the `scene.json`. For the sake of simplicity in this specification, we are assuming a minimum interface `{"main": "bin/scene.js"}` to illustrate how to load and run the code.

Reviewer: ```suggestion
As explained above, scenes are programs written in JavaScript. The deployed scenes must comply with the Scene schema defined in [ADR-51](/adr/ADR-51). Entities can be loaded as scenes if their metadata matches the `scene.json`. For the sake of simplicity in this specification, we are assuming a minimum interface `{"main": "bin/scene.js"}` to illustrate how to load and run the code.
```

> First the information of the entity to be executed as scene is fetched. As stated avobe, the entities contain a list of all files deployed and the metadata for that entity (scene.json).

Reviewer: ```suggestion
First, the information of the entity to be executed as the scene is fetched. As stated above, the entities contain a list of all files deployed and the metadata for that entity (scene.json).
```

  updateEntityRotation(rotation)
  await sendUpdatesToRenderer()
})
```

> TODO: explain how to hook to other runtime events

### Loading RPC modules in the scenes

The scene runtime exposes two asynchronous functions `loadModule` and `callRpc`. Those are the key entrypoint to communicate the scenes with other modules like the renderer. The exposed modules are defined in the [protocol repository](https://github.com/decentraland/protocol/blob/main/proto/decentraland/kernel/apis/engine_api.proto).

```ts
// loadModule instantiates a remote module
declare function loadModule(moduleName: string): Promise<ModuleDescriptor>

// the callRpc function takes a rpcHandle from loadModule result and a
// methodName from the same result, combined with a list of arguments
// it returns a promise executed in the remote module.
declare function callRpc(rpcHandle: string, methodName: string, args: any[]): Promise<ProxyModule>

declare type MethodDescriptor = { name: string }
declare type ModuleDescriptor = {
  rpcHandle: string
  methods: MethodDescriptor[]
}
```

> TODO: define and document naming conventions about code generation for modules

### Synchronizing scene's entities with the renderer

> The scenes synchronize with the renderer via the `EngineAPI.SendToRenderer` RPC using the CRDT protocol defined in [ADR-117](/rpc/ADR-117). The renderer will keep a local copy of all the entities and components required for rendering. Those components are in their majority serialized using protobuf as defined in [ADR-123](/rpc/ADR-123).

Reviewer: ```suggestion
The scenes synchronize with the renderer via the `EngineAPI.SendToRenderer` RPC using the CRDT protocol defined in [ADR-117](/rpc/ADR-117). The renderer will keep a local copy of all the entities and components required for rendering. Those components are mostly serialized using protobuf as defined in [ADR-123](/rpc/ADR-123).
```
